source/_integrations/homekit_controller.markdown

```diff
@@ -45,67 +45,79 @@ ha_integration_type: integration
+- [via Bluetooth](#adding-a-homekit-device-through-bluetooth)
+  - [migrate to a Thread network via Home Assistant](#adding-a-homekit-device-to-a-thread-network-via-home-assistant)
+  - [migrate to Thread network via Apple Thread border router](#adding-a-homekit-device-via-apple-thread-border-router)
 
-- Alarm Control Panel (HomeKit security system)
-- Climate (HomeKit thermostats and air conditioners)
-- Cover (HomeKit garage door openers, windows, or window coverings)
-- Light (HomeKit lights)
-- Lock (HomeKit lock)
-- Switch (HomeKit switches, outlets and valves)
-- Binary Sensor (HomeKit motion, contact, occupancy, carbon monoxide and smoke sensors)
-- Sensor (HomeKit humidity, temperature, co2 and light level sensors)
-- Fan
-- Air Quality
-- Humidifier (HomeKit humidifiers and dehumidifiers)
-- Automation Triggers (HomeKit 'stateless' accessories like buttons, remotes and doorbells)
+## Adding a HomeKit device via Ethernet or Wi-Fi
 
-<div class='note'>
+The HomeKit controller integration automatically detects HomeKit [compatible devices](#supported-devices) that are ready to pair if the [`zeroconf`](/integrations/zeroconf/) integration is enabled. The `zeroconf` is enabled by default on new installations via the [`default_config`](/integrations/default_config/) integration.
 
-  If your device is not on this list then you may still be able to pair it, and it will be visible in the Device Registry. But Home Assistant may not create entities for it.
+### Prerequisites
 
-</div>
+- If you do not have the [`default_config`](/integrations/default_config/) integration, add [`zeroconf`](/integrations/zeroconf/) to your configuration file.
+- Find your HomeKit pairing code. The code is on the device itself, or on the packaging. If your device has a screen, it may be shown on screen.
+   - There is no way to recover this if you do not have it. In this case, you will need to contact the manufacturer to see what options you have.
+- Make sure your device is powered up.
+- If your device is in your network but is paired with an Apple device via HomeKit, reset the device.
+   - Otherwise you won't be able to pair it with Home Assistant.
+   - HomeKit devices can only be paired to a single controller at once.
+- If your device is not already in your network, join the device to your network:
+   - Follow the instructions of the manufacturer on how to join the device to the network
+   - If you do not have the instructions, or the device is HomeKit only:
+     - Pair it with the Apple Home app (using the iOS Home app).
+     - After it has been paired with the Apple Home app, remove the device from the app again.
+     - Do not reset the device.
+     **Result**: Adding the device to the Home app and then removing it again has two effects:
+       - It adds the device to your network. It stays in the network even after you removed it from the app.
+       - Removing the device from the app opens it up for pairing with Home Assistant.
```

Comment:
This is kinda nice explanation what is going on. And the same applies to above. Can we maybe unify the two bullet points somewhat (the "If your device is in your network" and "If your device is not already in your network").

Reply:
How about moving the explanation out of the list so that it is clearer that it applies to both?
That way, we could keep both bullet points. They represent 2 different starting points. 

What do you think?

Reply:
Yeah, just make sure it's clear to the reader that it applies to both. We are anyways a bit too deeply nested here :sweat_smile:

Reply:
agreed, the nesting is a bit involved :smirk: